### ISSUE TITLE
local test failures in  org.eclipse.xtext.parser.indentation.tests when called gradle

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-core#608]
i have local test failures in  org.eclipse.xtext.parser.indentation.tests when called gradle.

**Comment by cdietrich** [comment]
```
com.google.inject.ProvisionException: Guice provision errors:

1) Error injecting constructor, org.eclipse.emf.common.util.WrappedException: java.lang.RuntimeException: Cannot create a resource for 'classpath:/org/eclipse/xtext/parser/indentation/IndentationAwareTestLanguage.xtextbin'; a registered resource factory is needed
  at org.eclipse.xtext.common.services.TerminalsGrammarAccess.<init>(TerminalsGrammarAccess.java:31)
  at org.eclipse.xtext.common.services.TerminalsGrammarAccess.class(TerminalsGrammarAccess.java:31)
  while locating org.eclipse.xtext.common.services.TerminalsGrammarAccess
    for parameter 1 at org.eclipse.xtext.parser.indentation.services.IndentationAwareTestLanguageGrammarAccess.<init>(IndentationAwareTestLanguageGrammarAccess.java:174)
  at org.eclipse.xtext.parser.indentation.services.IndentationAwareTestLanguageGrammarAccess.class(IndentationAwareTestLanguageGrammarAccess.java:174)
  while locating org.eclipse.xtext.parser.indentation.services.IndentationAwareTestLanguageGrammarAccess
  while locating org.eclipse.xtext.IGrammarAccess
    for field at org.eclipse.xtext.linking.lazy.LazyLinker.grammarAccess(LazyLinker.java:300)
  while locating org.eclipse.xtext.linking.lazy.LazyLinker
  while locating org.eclipse.xtext.linking.ILinker
    for field at org.eclipse.xtext.resource.XtextResource.linker(XtextResource.java:461)
  while locating org.eclipse.xtext.linking.lazy.LazyLinkingResource
  while locating org.eclipse.xtext.resource.XtextResource
Caused by: org.eclipse.emf.common.util.WrappedException: java.lang.RuntimeException: Cannot create a resource for 'classpath:/org/eclipse/xtext/parser/indentation/IndentationAwareTestLanguage.xtextbin'; a registered resource factory is needed
	at org.eclipse.xtext.parser.BaseEPackageAccess.loadResource(BaseEPackageAccess.java:57)
```

**Comment by cdietrich** [comment]
java main showing (maybe) the cause

```
package org.eclipse.xtext.parser.indentation;

import java.io.IOException;

import org.eclipse.emf.ecore.resource.Resource;
import org.eclipse.xtext.parser.indentation.tests.IndentationAwareTestLanguageInjectorProvider;
import org.eclipse.xtext.testing.IRegistryConfigurator;
import org.eclipse.xtext.testing.smoketest.internal.WrappingInjectorProvider;

import com.google.inject.Injector;

/**
 * @author dietrich - Initial contribution and API
 */
public class Main {
	
	public static void main(String[] args) throws IOException {
		IndentationAwareTestLanguageInjectorProvider ip = new IndentationAwareTestLanguageInjectorProvider();
		Injector i1 = ip.getInjector();
		((IRegistryConfigurator)ip).setupRegistry();
		System.out.println(Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().keySet());
		((IRegistryConfigurator)ip).restoreRegistry();
		System.out.println(Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().keySet());
		WrappingInjectorProvider wip = new WrappingInjectorProvider(ip);
		Injector i2 = wip.getInjector();
		((IRegistryConfigurator)wip).setupRegistry();
		System.out.println(Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().keySet());
		((IRegistryConfigurator)wip).restoreRegistry();
		System.out.println(Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().keySet());
	}

}

```

**Comment by cdietrich** [comment]
@szarnekow it would be nice if you could find the time to have a look

**Comment by cdietrich** [comment]
Fixed in 2.14
